lib/truffler.js

```diff
@@ -141,16 +141,21 @@ Truffler.prototype._run = function(url, options, done) {
 		// Open the URL in PhantomJS
 		openUrl: guardAgainstTimeout(function(next) {
 			options.log.debug('Opening "' + url + '" in PhantomJS');
+
+			state.page.onResourceError = function(resourceError) {
+				this.reason = resourceError.errorString;
```

Comment:
Yeah, it is a bit ugly. The problem we had with using `var` was that the scoping seemed to be wrong when the code ran in the tests, making `reason` always `undefined` there. I wonder if there's some tweaking of the test setup we could do to resolve this without needing `this`/`bind`?